src/utils.cairo

```diff
@@ -1,6 +1,50 @@
-use core::sha256::{compute_sha256_byte_array, compute_sha256_u32_array};
 use core::num::traits::{Zero, One, BitSize};
-use core::starknet::secp256_trait::Secp256PointTrait;
+use core::sha256::{compute_sha256_byte_array, compute_sha256_u32_array};
+
+#[derive(Copy, Drop, Debug)]
+// pub is required here as we define Hash in utils and we need to import Hash in merkle_tree.cairo
+// for arguments to merkle_root function
+pub struct Hash {
+    value: [u32; 8]
+}
+
+#[generate_trait]
+pub impl HashImpl of HashTrait {
+    #[inline(always)]
+    fn new() -> Hash {
```

Comment:
Shouldn't this take an Array<u32> as argument and calculate hash?

Reply:
I don't know? I first thought you wanted a new() function that just generates a default hash . That's what new() functions do generally? 
Woulndt it be an Into impl for Array to Hash?

Reply:
isn't this the constructor to create the hash from a [u32; 8]? I've used it like this
 fn new(value: [u32; 8]) -> Hash {
        Hash { value }
    }

Reply:
I see no use for default value of a hash. Maybe function to create a hash should have a different name?

Reply:
yes ok ! thanks!